server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/PathAndOptionalRequestConverter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.internal.api.converter;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.common.collect.Iterables;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.annotation.RequestConverterFunction;
+import com.linecorp.centraldogma.common.PathAndOptional;
+
+import io.netty.handler.codec.http.QueryStringDecoder;
+
+public class PathAndOptionalRequestConverter implements RequestConverterFunction {
+
+    private static final Splitter querySplitter = Splitter.on('&').trimResults().omitEmptyStrings();
+
+    private static final Splitter keyValueSplitter = Splitter.on('=').trimResults().omitEmptyStrings();
+
+    @Nullable
+    @Override
+    public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage request,
+                                 Class<?> expectedResultType) throws Exception {
+        final String queryString = ctx.query();
+        if (queryString != null) {
+            final String decodeString = QueryStringDecoder.decodeComponent(queryString);
```

Comment:
How about using `QueryStringDecoder.parameters()`?

Reply:
I have to parse it by myself because I have to keep the order of the `path` and   `optional_path`.
For example, when a user specify the query such as
```
path=1.json&optional_path=2.json&path=3.json
```
the aggregation will be applied in the order of `1.json`, `2.json`, `3.json`. But, it's hard to keep the original order if I use `parameters()`. :)